solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/JCSMPMessageSource.java

```diff
@@ -225,21 +229,33 @@ public void resume() {
 		Lock writeLock = readWriteLock.writeLock();
 		writeLock.lock();
 		try {
-			if (!isRunning()) {
-				logger.info(String.format("Can't resume message source %s as it is not running", id));
-				return;
-			}
+			logger.info(String.format("Resuming message source %s", id));
 			if (flowReceiverContainer != null) {
-				logger.info(String.format("Resuming message source %s", id));
-				flowReceiverContainer.resume();
+				try {
+					flowReceiverContainer.resume();
+				} catch (JCSMPException e) {
+					throw new RuntimeException(String.format("Failed to resume message source %s", id), e);
+				}
 			}
+			paused = false;
 		} finally {
 			writeLock.unlock();
 		}
 	}
 
 	@Override
 	public boolean isPaused() {
-		return flowReceiverContainer.isPaused();
+		if (paused) {
+			if (flowReceiverContainer.isPaused()) {
+				return true;
+			} else {
+				logger.warn(String.format(
+						"Flow receiver container %s is unexpectedly paused for message source %s",
```

Comment:
should this say: "Flow receiver container %s is unexpectedly **running** for message source %s "?

Reply:
Yup, it does. Look at the latest commit. I think you're looking at a diff that's one commit behind.